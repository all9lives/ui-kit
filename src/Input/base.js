import { css } from 'styled-components'

const base = css`
  width: 100%;
`

export default base
