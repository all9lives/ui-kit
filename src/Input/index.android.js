import styled from 'styled-components/native'
import base from './base'
import defaultProps from './defaultProps'

const typeMapping = {
  number: 'numeric',
  password: 'default',
  email: 'email-address'
}

const Input = styled.TextInput.attrs({
  secureTextEntry: props => props.type === 'password',
  keyboardType: props => (typeMapping[props.type] || props.type),
  editable: props => !props.readOnly
})`
  ${base}
`
Input.defaultProps = defaultProps

export default Input
