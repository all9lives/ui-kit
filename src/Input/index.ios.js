import Input from './index.android'

export default Input.extend`
  height: 60px;
  padding: 2px 6px;
`
