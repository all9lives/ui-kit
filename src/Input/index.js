import styled from 'styled-components'
import borderBox from '../border-box'
import base from './base'
import defaultProps from './defaultProps'

const Input = styled.input.attrs({
  onChange: (props) => e => {
    if (props.onChangeText) {
      props.onChangeText(e.target.value)
    }
    if (props.onChange) {
      props.onChange(e)
    }
  }
})`
  ${base}
  padding: 0.5rem 0.3rem;
`.extend`
  ${borderBox}
`
Input.defaultProps = defaultProps

export default Input
