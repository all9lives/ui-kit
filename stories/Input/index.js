import React from 'react'
import { action } from '@storybook/addon-actions'
import createStory from '../../storybook/createStory'
import Input from '../../src/Input'

createStory('Input', module)
  .add('with `placeholder:Your Name`', () => <Input placeholder='Your Name' />)
  .add('with `type:password`', () => <Input type='password' />)
  .add('with `type:number`', () => <Input type='number' />)
  .add('with `readOnly`', () => <Input readOnly />)
  .add('with `onChange`', () => <Input onChange={action('Input onChange')} />)
  .add('with `onChangeText`', () => <Input onChangeText={action('Input onChangeText')} />)
