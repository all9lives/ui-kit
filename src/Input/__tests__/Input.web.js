/* eslint-env jest */
import 'jest-styled-components'
import React from 'react'
import { shallow } from 'enzyme'
import Input from '../index.js'

describe('Input.web', () => {
  it('renders a snapshot', () => {
    const wrapper = shallow(
      <Input />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `placeholder:Placeholder`', () => {
    const wrapper = shallow(
      <Input placeholder='Placeholder' />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `type:password`', () => {
    const wrapper = shallow(
      <Input type='password' />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `readOnly`', () => {
    const wrapper = shallow(
      <Input readOnly />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `onChange`', () => {
    const wrapper = shallow(
      <Input onChange={() => {}} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `onChangeText`', () => {
    const wrapper = shallow(
      <Input onChangeText={() => {}} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('triggers onChange without passing functions', () => {
    const wrapper = shallow(
      <Input />
    )
    const result = wrapper.props().onChange()
    expect(result).toMatchSnapshot()
  })

  it('triggers onChange', () => {
    const wrapper = shallow(
      <Input onChange={() => 'onChanged'} />
    )
    const result = wrapper.props().onChange()
    expect(result).toMatchSnapshot()
  })

  it('triggers onChangeText', () => {
    const wrapper = shallow(
      <Input onChangeText={() => 'onChangedText'} />
    )
    const result = wrapper.props().onChange({ target: { value: 'onChangedText' } })
    expect(result).toMatchSnapshot()
  })
})
