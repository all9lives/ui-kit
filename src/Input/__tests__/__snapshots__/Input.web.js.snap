// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.web renders a snapshot 1`] = `
.c0 {
  width: 100%;
  padding: 0.5rem 0.3rem;
  box-sizing: border-box;
}

<input
  className="c0"
  onChange={[Function]}
  placeholder="Placeholder"
/>
`;

exports[`Input.web renders a snapshot with \`onChange\` 1`] = `
.c0 {
  width: 100%;
  padding: 0.5rem 0.3rem;
  box-sizing: border-box;
}

<input
  className="c0"
  onChange={[Function]}
  placeholder="Placeholder"
/>
`;

exports[`Input.web renders a snapshot with \`onChangeText\` 1`] = `
.c0 {
  width: 100%;
  padding: 0.5rem 0.3rem;
  box-sizing: border-box;
}

<input
  className="c0"
  onChange={[Function]}
  placeholder="Placeholder"
/>
`;

exports[`Input.web renders a snapshot with \`placeholder:Placeholder\` 1`] = `
.c0 {
  width: 100%;
  padding: 0.5rem 0.3rem;
  box-sizing: border-box;
}

<input
  className="c0"
  onChange={[Function]}
  placeholder="Placeholder"
/>
`;

exports[`Input.web renders a snapshot with \`readOnly\` 1`] = `
.c0 {
  width: 100%;
  padding: 0.5rem 0.3rem;
  box-sizing: border-box;
}

<input
  className="c0"
  onChange={[Function]}
  placeholder="Placeholder"
  readOnly={true}
/>
`;

exports[`Input.web renders a snapshot with \`type:password\` 1`] = `
.c0 {
  width: 100%;
  padding: 0.5rem 0.3rem;
  box-sizing: border-box;
}

<input
  className="c0"
  onChange={[Function]}
  placeholder="Placeholder"
  type="password"
/>
`;
